content/academy/puppeteer_playwright/page/interacting_with_a_page.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Interacting with a page
+description: Learn how to programmatically do actions on a page such as clicking, typing, and pressing keys. Also, discover a common roadblock that comes up when automating. 
+menuWeight: 1
+paths:
+    - puppeteer-playwright/page/interacting-with-a-page
+---
+
+# [](#interacting-with-a-page) Interacting with a page
+
+The **Page** object has whole boat-load of functions which can be used to interact with the loaded page. We're not going to go over every single one of them right now, but we _will_ use a few of the most common ones to add some functionality to our current project.
+
+Let's say that we want to automate searching for **hello world** on Google, then click on the first result and log the title of the page to the console, then take a screenshot and write it it to the filesystem. In order to understand how we're going to automate this, let's break down how we would do it manually:
+
+1. Click on the button which accepts Google's cookies policy (To see how it looks, open Google in an anonymous window.)
+2. Type **hello world** into the search bar
+3. Press **Enter**
+4. Wait for the results page to load
+5. Click on the first result
+6. Read the title of the result's page
+7. Screenshot the page
+
+Though it seems complex, the wonderful **Page** API makes all of these actions extremely easy to perform.
+
+## [](#clicking-and-pressing-keys) Clicking & pressing keys
+
+Let's first focus on the first 3 steps listed above. By using `page.click()` and the CSS selector of the element to click, we can click an element:
+
+```marked-tabs
+<marked-tab header="Playwright" lang="JavaScript">
+// Click the "I agree" button
+await page.click('button:has-text("I agree")');
+<marked-tab>
+<marked-tab header="Puppeteer" lang="JavaScript">
+// Click the "I agree" button
+await page.click('button + button:nth-child(2)');
+<marked-tab>
+```
```

Comment:
```suggestion
```marked-tabs
<marked-tab header="Playwright" lang="JavaScript">
// Click the "I agree" button
await page.click('button:has-text("I agree")');
</marked-tab>
<marked-tab header="Puppeteer" lang="JavaScript">
// Click the "I agree" button
await page.click('button + button:nth-child(2)');
</marked-tab>
```
```